app/uk/gov/hmrc/gform/controllers/helpers/FormDataHelpers.scala

```diff
@@ -52,22 +53,24 @@ object FormDataHelpers {
       .map(_.map { case (field, values) =>
         (
           field,
-          values.map { value =>
-            val matches = invisibleCharMatches(value)
-            if (matches.isEmpty) {
-              trimAndReplaceCRLFWithLF(value)
-            } else {
-              logger.info(
-                s"Found invisible characters in field $field. " +
-                  s"Matches are [${matches
-                    .map { case (m, count) =>
-                      s"${getUnicode(m)}:${getDesc(m)}($count)"
-                    }
-                    .mkString(", ")}]"
-              )
-              replaceInvisibleChars(value).trim
+          values
+            .map(StringEscapeUtils.unescapeHtml4)
```

Comment:
Why is this needed?

Reply:
We need to convert it from the `HTML Entity` to Unicode. For example; when a user enters  `&#2;`, it needs to be converted to Unicode to find in the INVISIBLE_UNICODE_DESC_MAPPING.

Reply:
It needs to be converted from HTML Entity(like `&#2;`) to Unicode(\u0002) to find it in the INVISIBLE_UNICODE_DESC_MAPPING

Reply:
What if I enter it as unicode char?

Reply:
It replaces both of them. unit test is [here](https://github.com/hmrc/gform-frontend/pull/1459/files#diff-9266775e40fad540ff83cfa8089a9689143a4302a83129759182c06bd1ad8d8eR251)

Reply:
Oh, now I see it. It's clever.